Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +193,74 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+
+            if (!IsValidCaseForInsertForNullOrEmpty(array, index))
+            {
+                return array;
+            }
+
+            if (! IsValidCaseForInsertForNullOrEmpty(array, number, index)  )
+            {
+                if (index == 0)
+                {
+                    return new [] { number };
+                }
+                return array; 
+            }
+
+           
+            
+            var newArray = Insert(array, number, index);
+
+            return newArray;
+
+        }
+
+        public static bool IsValidCaseForInsertForNullOrEmpty(int[] array, int index)
+        {
+            if (index >= 0)
+            {
+                return true;  
+            }
+            return false; 
+        } 
+           
+        
+            
+
+        public static bool  IsValidCaseForInsertForNullOrEmpty(int[] array, int number, int index)
```

Comment:
Same issue as above only with index and number.

Reply:
sorry, here is miss you with what you exactly wants

Reply:
I mean:
If you remove int[] array from line 216 and int number and int index from line 228 your functions will work the same because they do not use these parameters.
If you do so you have to change all calls of the functions, too, of course.

This would be 194 and 199 then. :)

Reply:
done, with replacing this logic into RemoveAt